src/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceController.cs

```diff
@@ -0,0 +1,802 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using INTPTR_INTCAST = System.Int32;
+using INTPTR_INTPTRCAST = System.IntPtr;
+using Microsoft.Win32.SafeHandles;
+using System.Text;
+using System.Runtime.InteropServices;
+using System.ComponentModel;
+using System.Diagnostics;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Globalization;
+using System.Security;
+
+namespace System.ServiceProcess
+{
+    /// This class represents an NT service. It allows you to connect to a running or stopped service
+    /// and manipulate it or get information about it.
+    public class ServiceController : IDisposable
+    {
+        private readonly string _machineName;
+        private readonly ManualResetEvent _waitForStatusSignal = new ManualResetEvent(false);
```

Comment:
Or, if Thread.Sleep doesn't exist in the profile, at least consider making _waitForStatusSignal static so that one instance of it isn't needed per object.